firmware/hackrf_usb/hackrf_usb.c

```diff
@@ -194,6 +194,49 @@ static void m0_rom_to_ram() {
 	}
 }
 
+void initial_tx_buffer_fill_completed(void* user_data, unsigned int transferred) {
+	(void)user_data;
+	(void)transferred;
+
+	// Now we have filled the first half of the transfer buffer we reset the buffer offset to zero
+	// so that we will transmit from the start of the buffer and to trigger the transfer of the
+	// second half of the buffer.
+	usb_bulk_buffer_offset = 0;
+
+	// The SGPIO output is double buffered. Therefore we must write two samples to the SGPIO
```

Comment:
I think this is the correct but could do with a second opinion.

Reply:
I don't think it does work like this: `REG` and `REG_SS` are swapped in a separate operation, writing to `REG_SS` doesn't trigger a swap.

Reply:
That makes sense. To be honest I just tried this, discovered it worked and didn't bother to think about why. But I now realise that I was measuring latency.

I had a device that outputs a square wave with a 10 second period. This was used as a hardware trigger for transmit. The HackRF was fed samples to generate a matching square wave and I compared the distance between the two edges. However, I did it on the rising edge after the rising edge that triggered transmit. So I wouldn't have seen if the initial samples were wrong.